src/Pages/LoginPage/Components/LoginForm.jsx

```diff
@@ -0,0 +1,136 @@
+    email: "",
+    password: "",
+  });
+
+  const loginInputHandler = (e) => {
+    setLoginDetails({
+      ...loginDetails,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const formSubmitHandler = async (e) => {
+    e.preventDefault();
+    const { email, password } = loginDetails;
+    setError("");
+    try {
+      const response = await axios.post("/api/auth/login", {
+        email,
+        password,
+      });
+      if (response.status === 200) {
+        setError(null);
+        authDispatch({ type: "LOGIN", payload: response.data });
+
+        //Save user info in local storage
+        localStorage.setItem("user", JSON.stringify(response.data));
+
+        //reset on submission
+        setLoginDetails({
+          email: "",
+          password: "",
+        });
+        if (pathname === "/products") {
+          setShowModal(false);
+        }
+        setShowAlert({
+          showAlert: true,
+          alertMessage: "Logged in successfully!",
+          type: "success",
+        });
+        if (pathname === "/login") {
+          navigate(-1);
+          //         setShowModal(false);
+        }
+      }
+    } catch (err) {
+      console.log(err);
+      if (err.response.status === 401) {
+        setError("Invalid email or password. Please try again.");
+      } else if (err.response.status === 404) {
+        setError("No user found with this email. Please try again.");
+      } else {
+        setError("Unexpected error. Please try again in some time.");
+      }
+    }
+  };
+
+  const testLogin = () => {
+    setLoginDetails({
+      ...loginDetails,
+      email: "adarshbalika@gmail.com",
+      password: "adarshBalika123",
+    });
+  };
+
+  const loginClasses = "login__container " + props.classes;
+
+  return (
+    <main className={loginClasses}>
+      <form
+        className="input-form"
+        autoComplete="on"
+        onSubmit={formSubmitHandler}
+      >
+        <h3>Login</h3>
+        <label htmlFor="email">
+          Email
+          <span className="red">*</span>
+        </label>
+        <input
+          type="text"
+          name="email"
+          className={error ? "input-error" : ""}
+          value={loginDetails.email}
+          placeholder="abc@gmail.com"
+          required
+          onChange={loginInputHandler}
+        />
+
+        <label className="label" htmlFor="password">
+          Password
+          <span className="red">*</span>
+        </label>
+        <div className="">
+          <input
+            type={showPassword ? "text" : "password"}
+            name="password"
+            className={"password-input" + (error ? " input-error" : "")}
+            value={loginDetails.password}
+            placeholder="*********"
+            required
+            onChange={loginInputHandler}
+          />
+          <span onClick={() => setShowPassword((prev) => !prev)}>
+            <i className="fa fa-eye" aria-hidden="true"></i>
+          </span>
```

Comment:
UI suggestion: The change of eye is left to show if the password is hidden or visible.
Also, I think you can directly use onClick on the `<i>` tag. Did you add span for CSS? do let me know.

Reply:
Thanks I missed the eye thing.
Yes, span is for the CSS styling.

Reply:
Got it